src/metaswitch/clearwater/queue_manager/test/test_fvtests.py

```diff
@@ -0,0 +1,165 @@
+# as those licenses appear in the file LICENSE-OPENSSL.
+
+from time import sleep
+import logging
+import sys
+import unittest
+import json
+import os
+from functools import partial
+from metaswitch.clearwater.etcd_tests.etcdcluster import EtcdCluster
+from ..etcd_synchronizer import EtcdSynchronizer
+from plugin import TestFVPlugin
+
+logging.getLogger().addHandler(logging.StreamHandler(sys.stderr))
+logging.getLogger().setLevel(logging.INFO)
+
+def wait_for_success_or_fail(sync, pass_criteria):
+    for x in range(300):
+        val = json.loads(sync._client.read("/clearwater/site1/configuration/queue_test").value)
+        if pass_criteria(val):
+            return True
+        sleep(1)
+    print "Queue config not updated as expected, final value was: ", val
+    return False
+
+def is_queue_config_as_expected(errors, queue, val):
+    return (errors == len(val.get("ERRORED"))) and \
+           (0 == len(val.get("COMPLETED"))) and \
+           (queue == len(val.get("QUEUED")))
+
+class QueueManagerTestBase(unittest.TestCase):
+    def tearDown(self):
+        for sync in self.syncs:
+            sync.terminate()
+        sleep(5)
+        self.c.delete_datadir()
+
+    def create_synchronizers(self, count):
+        self.c = EtcdCluster(count)
+
+        self.syncs = []
+        self.plugins = []
+
+        # Create a synchronizer for each node
+        x = 0
+        for ip in self.c.servers.values():
+            self.plugins.append(TestFVPlugin())
+            self.syncs.append(EtcdSynchronizer(self.plugins[x], ip._ip, 'site1', 'clearwater', 'nodetype%s' % x))
+            self.syncs[x].start_thread()
+            x += 1
+
+        sleep(10)
+        # Create another synchronizer to act like modify_nodes_in_queue
+        self.dummy_sync = EtcdSynchronizer(self.plugins[0], ip._ip, 'site1', 'clearwater', 'nodetypedummy')
+
+    def add_all_nodes_to_queue(self, force):
+        queue_list = ""
+        for node in self.syncs:
+            queue_list += "{\"ID\":\"%s\",\"STATUS\":\"QUEUED\"}," % node._id
+        self.dummy_sync._client.write("/clearwater/site1/configuration/queue_test", "{\"FORCE\": " + force + ", \"ERRORED\": [], \"COMPLETED\": [], \"QUEUED\": [" + queue_list[:-1] + "]}")
+
+    # Test that a two node queue is processed appropriately
+    @unittest.skipUnless(os.environ.get("SLOW"), "SLOW=T not set")
+    def dtest_two_node_queue(self):
+        self.create_synchronizers(2)
+        
+        # Set up the plugins
+        self.plugins[0].set_front_of_queue_callback(partial(self.dummy_sync.remove_from_queue, True, self.syncs[0]._id))
+        self.plugins[1].set_front_of_queue_callback(partial(self.dummy_sync.remove_from_queue, True, self.syncs[1]._id))
+        self.assertFalse(self.plugins[0].at_front_of_queue_called)
+        self.assertFalse(self.plugins[1].at_front_of_queue_called)
+
+        # Add nodes to queue
+        self.add_all_nodes_to_queue("false")
```

Comment:
Maybe worth using keyword arguments here so it's obvious what "false" is (an initial value?),